src/ngraph/pass/reshape_elimination.cpp

```diff
@@ -192,3 +191,127 @@ void pass::ReshapeElimination::construct_dot_transpose_pattern()
     auto m = make_shared<pattern::Matcher>(preshape, callback);
     this->add_matcher(m);
 }
+
+void pass::RecurrentReshapeElimination::construct_recurrent_reshape()
+{
+    Shape shape_op{3};
+    Shape shape_r{1, 3};
+
+    auto no_fan_out = [](std::shared_ptr<Node> n) {
```

Comment:
Looks like this one can combine with the on in concat elimination as a helper function.

Reply:
That might be tricky since this is a lambda function. Refactoring this small piece of code might not be effective.

Reply:
You could make a helper where you passed the arguments to be captured by the lambda and returned the lambda.

Reply:
Sounds good :). Will do that.